platforms/jvm/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/daemon/DaemonGroovyCompiler.java

```diff
@@ -117,12 +117,11 @@ protected DaemonForkOptions toDaemonForkOptions(GroovyJavaJointCompileSpec spec)
             File toolsJar = jvm.getToolsJar();
             if (toolsJar == null) {
                 String contextualMessage = String.format("The 'tools.jar' cannot be found in the JDK '%s'.", jvm.getJavaHome());
-                throw problemReporter.throwing(problemSpec -> problemSpec
+                throw problemReporter.throwing(new IllegalStateException(contextualMessage), problemSpec -> problemSpec
```

Comment:
💭Looking at this pattern I'm pretty certain we did the right decision. This looks much intentional.